src/query/vacuum.c

```diff
@@ -4178,7 +4178,19 @@ vacuum_data_load_and_recover (THREAD_ENTRY * thread_p)
 	   * in the log_Gl header. After a long session in SA_MODE, the vacuum_Data.last_page->data->blockid will
 	   * be outdated. Instead, SA_MODE updates log_Gl.hdr.vacuum_last_blockid before removing old archives.
 	   */
-	  vacuum_Data.set_last_blockid (MAX (log_Gl.hdr.vacuum_last_blockid, vacuum_Data.last_page->data->blockid));
+	  VACUUM_LOG_BLOCKID last_blockid;
+
+	  if (vacuum_Data.is_restoredb_session)
+	    {
+	      last_blockid = vacuum_Data.last_page->data->blockid;
+	      log_Gl.hdr.vacuum_last_blockid = 0;
```

Comment:
I have two suggestions.

1. How about using `logpb_last_complete_blockid()` to initiaizlize `log_Gl.hdr.vacuum_last_blockid`. It is normally  used to set a value to the variable with `log_Gl.append.prev_lsa.pageid`.
2. How about initializing it in `log_recovery_resetlog()`, which resets some variables in the log header when resetting log to a certain past time. It is called when time-specific(-d) restoredb or incomplete recovery occurred by a page corruption. It seems more natural and safer because some post/pre-processing seems to be done here like `logpb_flush_header (thread_p);`

My suggestion is basically to handle the problem when it occurs, not it is used or detected.

Reply:
vacuum_Data.m_last_blockid is a low bound used to find log blocks that could not be added to the vacuum data file. If vacuum_Data.last_page->data->blockid < block id of log_Gl.append.prev_lsa.pageid, a log block that cannot be found may occur.

Reply:
Modified to initialize log_Gl.hdr.vacuum_last_blockid using log_recovery_resetlog() function. Thanks for the good comments.